code/src/components/DirectionButtons.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components';
+
+// ----------------------------------------------------------------
+
+export const DirectionButtons = ({ direction, action }) => {
+  const dispatch = useDispatch();
+  return (
+    <DirectionButton
+      onClick={(event) => dispatch(action(event))}
+      direction={direction}
+    >
+      <Arrow role="img" aria-label={direction + `-arrow`} direction={direction}>
+        ➤
+      </Arrow>
+      <DirectionText direction={direction}>Go {direction}</DirectionText>
+    </DirectionButton>
+  );
+};
+
+// ----------------------------------------------------------------
+
+const Arrow = styled.span`
+  display: inline-block;
+  font-size: 30px;
+  color: #fff;
+  transform: ${(props) =>
+    props.direction === 'North'
+      ? `rotate(-90deg)`
+      : props.direction === 'East'
+      ? `rotate(0deg)`
+      : props.direction === 'South'
+      ? `rotate(90deg)`
+      : `rotate(180deg)`};
+  margin-top: ${(props) => (props.direction === 'South' ? '10px' : '')};
+  margin-bottom: ${(props) => (props.direction === 'North' ? '10px' : '')};
```

Comment:
Neat.  How would this work if there were suddenly 8 directions?   Can this implementation be generalized to support such a scenario?